### PR TITLE
{cae}[intel/2020a] OpenFOAM v2006

### PR DESCRIPTION
**Pull request by ekieffer**
(created using `eb --new-pr`)

edit: requires ~~#11366~~ (CGAL)

### CODE REVIEW

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in this PR)
node3402.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/39be44cb9c5d962bd6526ed16f68ce5c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in this PR)
node2706.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/00c640414b83fa41d1fb81db50cb70a9 for a full test report.

**Comment by zao** [comment]
Test report by @zao
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
freja - Linux Ubuntu 20.04, x86_64, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 3.8.2
See https://gist.github.com/aacb5cb2e482157f42d6ab1bde8a681c for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
@ekieffer It looks like this need a sync with the current `develop` branch to make the tests happy.

Can you please run `eb --sync-pr-with-develop 11367`?

**Review by furstj — CHANGES_REQUESTED** [review]
Please remove OpenFOAM-v1906-wmake-ompi.patch. The patch is specific to OpenMPI and does not work for Intel MPI. Moreover, it breaks compilation of paraview plugins with intel MPI.

**Review by furstj — CHANGES_REQUESTED** [review]
OpenFOAM v2006 has comilation issues with intel compiler. Compilation stops with internal compiler error when compiling setAlphaField. 

It was discussed here:
https://develop.openfoam.com/Development/openfoam/-/issues/1821
and a patch was created by L. Gamet.

Pleas include L. Gamet's patch (you can find [here](https://develop.openfoam.com/Development/openfoam/-/issues/1821) )

**Comment by boegel** [comment]
@furstj The tests reports show that there are no compilation issues with `intel/2020a`?

**Comment by boegel** [comment]
(close/re-open to re-trigger tests)

File: easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v2006-intel-2020a.eb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+]
+
+dependencies = [
+    ('libreadline', '8.0'),
+    ('ncurses', '6.2'),
+    # OpenFOAM requires 64 bit METIS using 32 bit indexes (array indexes)
+    ('METIS', '5.1.0'),
+    ('SCOTCH', '6.0.9'),
+    ('CGAL', '4.14.3', '-Python-3.8.2'),
+    ('ParaView', '5.8.0', '-Python-3.8.2-mpi'),
```
Comment (boegel) [inline easybuild/easyconfigs/o/OpenFOAM/OpenFOAM-v2006-intel-2020a.eb:32]: @ekieffer `gnuplot` dependency is missing (see #11568)

```suggestion
    ('ParaView', '5.8.0', '-Python-3.8.2-mpi'),
    ('gnuplot', '5.2.8'),
```

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3501.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/43666dbcb7fd4dfb544e62d0cdad81eb for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3119.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/c96ba5ac6b6106fa72ec41665920f5f9 for a full test report.

**Comment by furstj** [comment]
@boegel there is still an issue with compilation with intel/2020a (at least at my computer). The compiler fails to compile "setAlphaField". The "setAlphaField" is not included in sanity checks so it looks like it's ok but it isn't. Please check also 
EBROOT/software/OpenFOAM/v2006-intel-2020a/OpenFOAM-v2006/log.linux64IccDPInt32Opt 
and try to find "setAlphaField" there.

We should probably include "setAlphaField" into sanity checking.

**Comment by boegel** [comment]
Indeed, I'm finding this in the log:

```
mpiicpc -cxx="icpc" -pthread -fp-trap=common -fp-model precise -DOPENFOAM=2006 -DWM_DP -DWM_LABEL_SIZE=32 -Wall -Wextra -Wnon-virtual-dtor -Wno-unused-parameter -Wno-invalid-offsetof -Wno-unknown-pragmas -diag-disable 327,654,1125,1292,2289,2304,11062,11074,11076  -O2 -xHost -ftz -fp-speculation=safe -fp-model source -std=c++11 -no-prec-div  -DNoRepository  -iquote. -IlnInclude -I/software/OpenFOAM/v2006-intel-2020a/OpenFOAM-v2006/src/OpenFOAM/lnInclude -I/software/OpenFOAM/v2006-intel-2020a/OpenFOAM-v2006/src/OSspecific/POSIX/lnInclude   -fPIC -c implicitFunctions/sin/sinImplicitFunction.C -o /software/OpenFOAM/v2006-intel-2020a/OpenFOAM-v2006/build/linux64IccDPInt32Opt/applications/utilities/preProcessing/setAlphaField/alphaFieldFunctions/implicitFunctions/sin/sinImplicitFunction.o                                                       
Internal error loop: assertion failed: find_seq_in_lookup_table: seq_number not found (shared/cfe/edgcpfe/il.c, line 4293)
compilation aborted for implicitFunctions/sin/sinImplicitFunction.C (code 4)
make[2]: *** [/software/OpenFOAM/v2006-intel-2020a/OpenFOAM-v2006/build/linux64IccDPInt32Opt/applications/utilities/preProcessing/setAlphaField/alphaFieldFunctions/implicitFunctions/sin/sinImplicitFunction.o] Error 4                                                                                                                                       
make[1]: *** [setAlphaField] Error 2
make[1]: *** Waiting for unfinished jobs....
```

So `platforms/linux64GccDPInt32Opt/bin/setAlphaField` is missing in the installation...

This looks like an internal compiler error though, so I'm not sure there's an easy way to fix this...

@olesenm Thoughts on this?

**Comment by branfosj** [comment]
The patch at https://develop.openfoam.com/Development/openfoam/-/commit/719d3f77bea23bbdb9c530ee2dde09383f03bc71.patch is for this issue.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3103.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/73fb15696951b8a6da134066e12b45d9 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3518.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/f47207f4348b53976a1b3fd901f8a516 for a full test report.

**Comment by boegel** [comment]
Those last test reports aren't very useful, patch is still missing 😅

**Review by boegel — CHANGES_REQUESTED** [review]
@ekieffer Please include the changes from https://github.com/ULHPC/easybuild-easyconfigs/pull/4 in here (just merging my PR to your branch should be sufficient)

**Comment by olesenm** [comment]
@boegel 
Would it not make more sense to simply bump the 2006 version to the upstream patched version?

https://sourceforge.net/projects/openfoam/files/v2006/OpenFOAM-v2006_201012.tgz

It definitely has the fix already in it.

FWIW: the OpenFOAM _master_ and _maintenance_ branches (and any resulting tar files) are very conservative. They are bugfix only, no new features.

**Comment by boegel** [comment]
@olesenm We could, but then we would version it as `2006-201012` or something like that, to discrimininate from a "vanilla" `2006`...

**Comment by olesenm** [comment]
> @olesenm We could, but then we would version it as `2006-201012` or something like that, to discrimininate from a "vanilla" `2006`...

Yes that would be the best. IMO much better than adding in EasyBuild patches onto the vanilla 2006, but that's just my opinion.

**Comment by boegel** [comment]
@boegelbot please test @ generoso
CORE_CNT=16

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
node3100.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/0f88e695d370c584c218d896b8aa7e5e for a full test report.

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
Going in, thanks @ekieffer!

**Pull request merged**